tools/CheckSignature.ps1

```diff
@@ -0,0 +1,114 @@
+<#
+    Check (and recurse) current directory     .\CheckSignature.ps1
+    Check directory after MSI install         .\CheckSignature.ps1 -MsiInstall
+    Check directory after gallery install     .\CheckSignature.ps1 -GalleryInstall
+#>
+[CmdletBinding(DefaultParameterSetName="CurrentDirectory")]
+Param
+(
+    [Parameter(ParameterSetName="MsiInstall", Mandatory=$true)]
+    [switch]$MsiInstall,
+    [Parameter(ParameterSetName="GalleryInstall", Mandatory=$true)]
+    [switch]$GalleryInstall,
+    [Parameter(ParameterSetName="CustomPath", Mandatory=$true)]
+    [string]$CustomPath
+)
+
+function Check-StrongName {
+    [CmdletBinding()]
+    param([Parameter(ValueFromPipeline=$true)][string]$path)
+    $output = & "sn.exe" -vf $path
```

Comment:
We should check for the existence of sn.exe on the path and fail fast if it is not found at the beginning of the Check-All script.